weechat-relay/src/main/java/com/ubergeek42/weechat/relay/RelayConnection.java

```diff
@@ -45,6 +45,10 @@
     IConnection conn;
     LinkedBlockingQueue<String> outbox = new LinkedBlockingQueue<String>();
 
+    // Tri-state boolean
+    private Boolean messageReceivedOnCurrentConnection = null;
```

Comment:
This field is being modified and accessed across threads; it should be `volatile`.